effect/shared/src/main/scala/scalaz/effect/Async.scala

```diff
@@ -32,7 +30,8 @@ object Async {
    *
    * See `IO.async0` for more information.
    */
-  final def later[E, A]: Async[E, A] = Later()
+  final def later[E, A]: Async[E, A] =
+    MaybeLater(_ => ())
```

Comment:
This will allocate a closure every single time, which will negatively affect performance in a measurable way.

To avoid this, replace with:

```scala
... = MaybeLater(Canceler)
```

Reply:
👍 , done